hedera-node/hedera-app/src/main/java/com/hedera/node/app/ServicesMain.java

```diff
@@ -107,7 +107,10 @@ MerkleHederaState newMerkleHederaState(
         final var migration = migrationFactory.apply(servicesSemVer);
         return new MerkleHederaState(
                 migration,
-                (event, metadata, provider) -> metadata.app().eventExpansion().expandAllSigs(event, provider),
+                (event, metadata, immutableState) -> {
+                    final var metaApp = metadata.app();
```

Comment:
Might be better to cast here rather than the line below.
Also, are we sure this will always cast to HederaApp?